docs/tutorials/htcondor.md

```diff
@@ -0,0 +1,189 @@
+deploy your HPC cluster. These will be caught when you perform `terraform apply`
+but you can save time by enabling them now by running:
+
+<walkthrough-enable-apis apis="storage.googleapis.com,compute.googleapis.com,secretmanager.googleapis.com"></walkthrough-enable-apis>
+
+## Build the Toolkit Binary
+
+To build HPC Toolkit binary from source run:
+
+```bash
+make
+```
+
+You should now have a binary named ghpc in the current directory. To verify the
+build run:
+
+```bash
+./ghpc --version
+```
+
+This should show you the version of the HPC Toolkit you are using.
+
+(Optional) To install the `ghpc` binary in your home directory under bin,
+run the following command:
+
+```bash
+make install
+exec $SHELL -l
+```
+
+## Generate a Deployment
+
+To create a deployment, an input blueprint file needs to be written or adapted
+from one of the examples found in the `examples/` or `community/examples`
+directories.
+
+This tutorial will use community/examples/htcondor-pool.yaml, which provisions
+a basic auto-scaling HTCondor pool.
+
+* a new VPC network secured from the public internet
+* an HTCondor Access Point for users to submit jobs
+* an HTCondor Central Manager that will operate the pool
+* a Managed Instance Group to scale a pool of HTCondor Execute Points to serve
+  new jobs as they are submitted
+
+The blueprint community/examples/htcondor-pool.yaml should be open in the Cloud
+Shell Editor (on the left).
+
+This file describes the cluster you will deploy. After you have inspected the
+file, use the ghpc binary to create a deployment directory by running:
+
+```bash
+./ghpc create community/examples/htcondor-pool.yaml --vars "project_id=<walkthrough-project-id/>"
+```
+
+> **_NOTE:_** The `--vars` argument is used to override `project_id` in the
+> blueprint variables. The `--vars` argument supports comma-separated list of
+> name=value variables to override blueprint variables. This feature only
+> supports variables of string type.
+
+This will create a deployment directory named `htcondor-001/`, which
+contains the terraform needed to deploy your cluster.
+
+## Deploy the Cluster
+
+Use the following commands to run terraform and deploy your cluster.
+
+```bash
+terraform -chdir=htcondor-001/htcondor init
+terraform -chdir=htcondor-001/htcondor validate
+terraform -chdir=htcondor-001/htcondor apply -auto-approve
+```
+
+The deployment will take about 3 minutes. There should be regular status updates
+in the terminal. If the `apply` is successful, a message similar to the following
+will be displayed:
+
+<!-- Note: Bash blocks give "copy to cloud shell" option.  -->
+<!-- "shell" or "text" is used in places where command should not be run in cloud shell. -->
+
+```shell
+Apply complete! Resources: xx added, 0 changed, 0 destroyed.
+```
+
+## Run a Job on the HTCondor pool
+
+Once the cluster has successfully been deployed, take the following steps to
+run a job:
```

Comment:
Suggestion: instead of automatically copying `hello-world.sub` into the access point via Terraform, have the user copy it over with `gcloud compute scp`. This will make it clear how they can copy their own `.sub` files over once they're ready to try their own jobs.

Reply:
To document a conversation: a goal is here is to have something that works without any authentication setup and also provides an example of how to automatically stage workflows to a pool. That doesn't mean there isn't value in understanding how to stage it in _with authentication_ but we want to the scope of the tutorial to be quick, easy and reliable.